src/Extensions/Hosting.Services.Web/WebEndpointInfo.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System.Globalization;
+using Microsoft.Omex.Extensions.Abstractions;
+
+namespace Microsoft.Omex.Extensions.Hosting.Services.Web
+{
+	/// <summary>
+	/// Endpoint description for sf web listener
+	/// </summary>
+	public class WebEndpointInfo
+	{
+		/// <summary>
+		/// Endpoint name
+		/// </summary>
+		public string Name { get; }
+
+		/// <summary>
+		/// Setting name that will contain common name of the certificate for https
+		/// </summary>
+		public string? SettingForCertificateCommonName { get; }
+
+		/// <summary>
+		/// Endpoint port
+		/// </summary>
+		public int Port { get; }
+
+		/// <summary>
+		/// Use https instead of http
+		/// </summary>
+		public bool UseHttps { get; }
+
+		private WebEndpointInfo(string endpointName, string? settingForCertificateCommonName)
+		{
+			Name = Validation.ThrowIfNullOrWhiteSpace(endpointName);
+			SettingForCertificateCommonName = settingForCertificateCommonName;
+			UseHttps = settingForCertificateCommonName != null;
```

Comment:
HTTP still used in multiple places by current services that's why it's present here.
For user, it would be a choice of using `WebEndpointInfo.CreateHttp` or `WebEndpointInfo.CreateHttps`, the constructor is private. Or you would prefer another way?

Reply:
No, that's okay would you open bugs for any existing services that you are aware of which are using HTTP though?

Reply:
Okay, I'll check the one that I remember